src/lib/transition_router/transition_router.ml

```diff
@@ -184,24 +187,19 @@ module Make (Inputs : Inputs_intf) :
         , transition_frontier_controller_writer ) =
       start_transition_frontier_controller ~verified_transition_writer
         ~clear_reader ~collected_transitions:[]
-        (Mvar.peek_exn frontier_mvar)
+        (peek_exn frontier_shared_mvar)
     in
     let controller_type =
       Broadcaster.create
         ~init:
           (`Transition_frontier_controller
-            ( Mvar.peek_exn frontier_mvar
+            ( peek_exn frontier_shared_mvar
```

Comment:
In this code, I created `Broadcaster` to swap between transition_frontier_controller mode and bootstrap mode. The Broadcast_pipe also achieves this, but it seems to be a cleaner. Can you replace `Broadcaster` in terms of `Broadcast_pipe`?

Reply:
As discussed, we're leaving this be for now since `Broadcaster` has synchronous semantics